lib/convection/model/template.rb

```diff
@@ -263,7 +263,12 @@ def render(stack_ = nil)
           'Parameters' => parameters.map(&:render),
           'Mappings' => mappings.map(&:render),
           'Conditions' => conditions.map(&:render),
-          'Resources' => all_resources.map(&:render),
+          'Resources' => all_resources.map do |resource|
+            if retain && resource.deletion_policy.nil?
+              resource.deletion_policy('Retain')
+            end
+            resource.render
```

Comment:
🎨 Add a newline above this to separate the conditional stanza and this return value.